openedx/core/djangoapps/discussions/docs/decisions/0006-discussions-api-fetch-inactive-providers.rst

```diff
@@ -0,0 +1,150 @@
+when switching providers between the in-built/legacy providers and LTI
+providers.
+
+The current mechanism for configuring providers in the course authoring MFE is
+through two steps. In the first step, you select a provider, and in the second
+step you change its configuration.
+
+However when changing providers, the only information you have from the API
+is about the current provider, meaning that you cannot have configuration
+pre-filled in the configuration page in the second step. The settings page will
+only have default information, which means that on saving the configuration in
+the second step, the original saved configuration will be overridden, and
+potentially might be in an inconsistent state.
+
+Requirements
+------------
+
+Split the configuration API into two parts where one part returns static
+data about the available providers, and the other can be the configuration
+data for the discussions provider.
+
+The configuration API should provide a way to fetch settings for a provider
+without needing to switch to it first.
+
+Alternatives Considered
+-----------------------
+
+Another way around the second issue above is to save the provider change
+first, fetch the configuration data and then proceed with the second
+configuration step.
+
+However this can be very disruptive since changing the provider first will
+trigger a course republish and a lot of background machinery to build
+discussion topics, and will change the live experience using old settings that
+the course author might have wanted to change before publishing.
+
+Decision
+--------
+
+We can split the existing configuration API into two parts. The first part will
+be a read-only API for listing provider details. No data in this API can be
+modified by the user.
+
+The other API will return all the writable configuration for the providers.
+It will support updating the settings of the current provider as it does
+now, but will also support returning the configuration/settings of other
+providers by taking a ``provider_id`` query parameter.
+
+
+Provider Endpoint
+~~~~~~~~~~~~~~~~~
+
+``/api/discussions/v0/course/{course_id}/providers``
+
+This will return the list of available providers and their data. This will
+return the following pieces of information:
+
+- ``active``: the active provider
+- ``features``: feature support levels
+- ``available``: provider listing
+
+**Sample response:**
+
+.. code-block:: json
+
+    {
+        "active": "legacy",
+        "features": [
+            {
+                "id": "primary-discussion-app-experience",
+                "feature_support_type": "basic"
+            },
+            …
+        ],
+        "available": {
+            "legacy": {
+                "features": [ … ],
+                "supports_lti": false,
+                "external_links": { … },
+                "messages": [ … ],
+                "has_full_support": true
+            },
+            …
+        }
```

Comment:
I am thinking we should pass the value of `pii_sharing_allowed_flag` to the frontend for read-only and decide whether to show PII settings to the user or not. This will simplify the flow on the front end as well. 

This setting is different from actual read-write pii_sharing_email and username. 
What do you think?

Reply:
I don't want to overload this ADR with other changes. Currently it's focussed on making it possible to preview provider settings. We can make other structural changes in other ADRs and PRs.

Reply:
Ok sure.